src/backend/marsha/core/tests/test_lti.py

```diff
@@ -581,3 +582,84 @@ def test_get_consumer_site_verified(self, mock_verify):
+            oauth_consumer_key="W1PWSNDUL7T7YGMCOWZH",
+        )
+        lti_parameters = {
+            "content_items": (
+                '{"@context":"http://purl.imsglobal.org/ctx/lti/v1/ContentItem",'
+                '"@graph":[{'
+                '"@type":"LtiLinkItem",'
+                '"url":"/lti/videos/0c02115f-dae7-4c05-b931-2ddc7e87abc1",'
+                '"title":"example.com-df8",'
+                '"frame":[]'
+                "}]"
+                "}"
+            ),
+            "oauth_version": "1.0",
+            "oauth_nonce": "557b295f193999bb0c515e332d77ecb9",
+            "oauth_timestamp": "1616586282",
+            "oauth_consumer_key": "W1PWSNDUL7T7YGMCOWZH",
+            "user_id": "2",
+            "lis_person_sourcedid": "",
+            "roles": (
+                "Instructor,"
+                "urn:lti:sysrole:ims/lis/Administrator,"
+                "urn:lti:instrole:ims/lis/Administrator"
+            ),
+            "context_id": "2",
+            "context_label": "aui",
+            "context_title": "aui",
+            "context_type": "CourseSection",
+            "lis_course_section_sourcedid": "",
+            "launch_presentation_locale": "en",
+            "ext_lms": "moodle-2",
+            "tool_consumer_info_product_family_code": "moodle",
+            "tool_consumer_info_version": "2020110902.01",
+            "oauth_callback": "about:blank",
+            "lti_version": "LTI-1p0",
+            "lti_message_type": "ContentItemSelectionRequest",
+            "tool_consumer_instance_guid": "localhost",
+            "tool_consumer_instance_name": "",
+            "tool_consumer_instance_description": "",
+            "accept_media_types": "application/vnd.ims.lti.v1.ltilink",
+            "accept_presentation_document_targets": "frame,iframe,window",
+            "accept_copy_advice": "false",
+            "accept_multiple": "true",
+            "accept_unsigned": "false",
+            "auto_create": "false",
+            "can_confirm": "false",
+            "lti_select_form_action_url": (
+                "http://localhost:8080/mod/lti/contentitem_return.php"
+                "?course=2&id=1&sesskey=BJ9rCsvIPs"
+            ),
+            "title": "localhost.marsha",
+            "text": "",
+            "oauth_signature_method": "HMAC-SHA1",
+            "oauth_signature": "5jUg6zMZLHOOPN+2FkiqBXz4uSw=",
+        }
+
+        request = self.factory.post(
+            reverse("select_lti_view"),
+            lti_parameters,
+            HTTP_REFERER="http://localhost:8080/mod/lti/contentitem.php",
+        )
+
+        lti = LTI(request)
+
+        sent_lti_parameters = lti_parameters
+        sent_lti_parameters["oauth_signature"] = "ydDCOlhFq8a2bGLtz9ebq+PHyYM="
+        self.assertDictEqual(
+            sent_lti_parameters,
+            lti.sign_post_request(
+                lti_parameters.get("lti_select_form_action_url"), lti_parameters
+            ),
+        )
```

Comment:
I don't understand how you can know the oauth_signature in advance without mocking time... ("5jUg6zMZLHOOPN+2FkiqBXz4uSw=") as its value depends on the time at which you calculate it. Remember the tests we did in Richie?

Afterthought I think this test does not work also because when you do:
```python
sent_lti_parameters = lti_parameters
```
you are copying the dictionary by its reference and both objects are then the same... Also I don't think the initial lti_parameters should have the oauth parameters and especially the signature.

Reply:
The inital lti_parameters are comes from the request made by the LTI consumer, so it contains all oauth parateters.

Reply:
Yes but this test does not work. It is impossible that you know the signature in advance... the dict assignation is done by reference not by value.
I'm not able to tell you how it should be done without diving in the code but I think it is fishy and not testing what you think it is :smile: